activerecord/lib/active_record/secure_token.rb

```diff
@@ -26,8 +26,11 @@ module ClassMethods
       def has_secure_token(attribute = :token)
         # Load securerandom only when has_secure_key is used.
         require 'active_support/core_ext/securerandom'
-        define_method("regenerate_#{attribute}") { update! attribute => self.class.generate_unique_secure_token }
-        before_create { self.send("#{attribute}=", self.class.generate_unique_secure_token) }
+        unless self.method_defined?("generate_#{attribute}")
```

Comment:
I might have been crazy, I thought we ended up having to add that for enums because inheritance or modules or something.. but I couldn't find it. Seems fine to remove it. If I figured out what it was, I can add it back with a corresponding test case.